src/models/TreeData.ts

```diff
@@ -83,11 +83,11 @@ export class TreeDataSource {
         }
     }
 
-    private isDataSource(input: TreeNode | TreeDataSource | Array<TreeNode>): input is TreeDataSource {
+    private isDataSource(input: TreeNode | TreeDataSource | Array<any>): input is TreeDataSource {
         return (<TreeDataSource>input).updateNode !== undefined;
     }
 
-    private isRootNodesArray(input: TreeNode | TreeDataSource | Array<TreeNode>): input is Array<TreeNode> {
+    private isRootNodesArray(input: TreeNode | TreeDataSource| Array<any>): input is Array<any> {
         return (<Array<TreeNode>>input).slice !== undefined;
```

Comment:
@vinkobedek 
Doesn't this cast negate the Array<any> above? If you send Array of type that isnt TreeNode it will return that root node is not array

Reply:
in the end, this is all just javascript, so this cast will succeed in runtime. this is why we have the need for these utility functions in the first place that determine of which object type it actually is. But it is definitely misleading, and i shall change it